crates/core_arch/src/x86/avx512bf16.rs

```diff
@@ -0,0 +1,1140 @@
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=1769,1651,1654,1657&avx512techs=AVX512_BF16&text=_mm512_cvtne2ps_pbh)
+#[inline]
+#[target_feature(enable = "avx512bf16,avx512f")]
+#[cfg_attr(test, assert_instr("vcvtne2ps2bf16"))]
+pub unsafe fn _mm512_cvtne2ps_pbh (a: __m512, b: __m512) -> __m512bh {
+    transmute(cvtne2ps2bf16_512(a.as_f32x16(), b.as_f32x16()))
+}
+
+/// Convert packed single-precision (32-bit) floating-point elements in two vectors 
+/// a and b to packed BF16 (16-bit) floating-point elements, and store the results 
+/// in single vector dst using writemask k (elements are copied from src when the 
+/// corresponding mask bit is not set).
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=1769,1651,1654,1657&avx512techs=AVX512_BF16&text=_mm512_mask_cvtne2ps_pbh)
+#[inline]
+#[target_feature(enable = "avx512bf16,avx512f")]
+#[cfg_attr(test, assert_instr("vcvtne2ps2bf16"))]
+pub unsafe fn _mm512_mask_cvtne2ps_pbh (src: __m512bh, k: __mmask32, a: __m512, b: __m512) -> __m512bh {
+    let cvt = _mm512_cvtne2ps_pbh(a, b).as_u16x32();
+    transmute(simd_select_bitmask(k, cvt, src.as_u16x32()))
+}
+
+/// Convert packed single-precision (32-bit) floating-point elements in two vectors 
+/// a and b to packed BF16 (16-bit) floating-point elements, and store the results 
+/// in single vector dst using zeromask k (elements are zeroed out when the corresponding 
+/// mask bit is not set).
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=1769,1651,1654,1657&avx512techs=AVX512_BF16&text=_mm512_maskz_cvtne2ps_pbh)
+#[inline]
+#[target_feature(enable = "avx512bf16,avx512f")]
+#[cfg_attr(test, assert_instr("vcvtne2ps2bf16"))]
+pub unsafe fn _mm512_maskz_cvtne2ps_pbh (k: __mmask32, a: __m512, b: __m512) -> __m512bh {
+    let cvt = _mm512_cvtne2ps_pbh(a, b).as_u16x32();
+    let zero = _mm512_setzero_si512().as_u16x32();
+    transmute(simd_select_bitmask(k, cvt, zero))
+}
+
+/// Convert packed single-precision (32-bit) floating-point elements in a to packed BF16 (16-bit) 
+/// floating-point elements, and store the results in a 128-bit wide vector.
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=1769,1651,1654,1657,1660&avx512techs=AVX512_BF16&text=_mm_cvtneps_pbh)
+#[inline]
+#[target_feature(enable = "avx512bf16,avx512vl")]
+#[cfg_attr(test, assert_instr("vcvtneps2bf16"))]
+pub unsafe fn _mm_cvtneps_pbh (a: __m128) -> __m128bh {
+    cfg_if! {
+        if #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] {
+            let mut result:__m128bh;
+            asm!(
+                    "vcvtneps2bf16 {0}, {1}",
+                    in(xmm_reg) a,
+                    lateout(xmm_reg) result,
+                    options(att_syntax)
+                );
+            result
+        }
+        else {
+            unreachable!()
+        }
+    }
```

Comment:
```suggestion
    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
    {
        let mut result:__m128bh;
        asm!(
                "vcvtneps2bf16 {0}, {1}",
                in(xmm_reg) a,
                lateout(xmm_reg) result,
                options(att_syntax)
        );
        result
    }
    #[cfg(not(any(target_arch = "x86", target_arch = "x86_64")))]
    {
        unreachable!()
    }
```

This should work instead of cfg-if.

Reply:
Thank you very much. It works. :)